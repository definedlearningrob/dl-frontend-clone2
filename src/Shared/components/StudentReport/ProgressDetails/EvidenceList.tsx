import { Trans } from 'react-i18next';
import { isEmpty, noop } from 'lodash-es';

import { EvidencesListItem } from '@shared/components/EvidenceListItem/EvidencesListItem';
import { TEvidence } from '@shared/resources/types';

type Props = {
  evidence: TEvidence[];
};

export const EvidenceList = ({ evidence }: Props) => {
  if (isEmpty(evidence)) {
    return null;
  }

  return (
    <div className='flex flex-col gap-xxs xxxl:gap-xs'>
      <h5 className='mb-0 text-xxs leading-base uppercase'>
        <Trans
          components={{
            neutralText: <span className='text-neutral-600' />,
          }}
          i18nKey='studentGoalReport.evidenceCount'
          values={{ count: evidence.length }}
        />
      </h5>
      <ul>
        {evidence.map((evidenceItem) => (
          <li key={evidenceItem.id} className='mx-sm my-x'>
            <EvidencesListItem
              {...evidenceItem}
              interactive={false}
              onDelete={noop}
              onNavigate={noop}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};
