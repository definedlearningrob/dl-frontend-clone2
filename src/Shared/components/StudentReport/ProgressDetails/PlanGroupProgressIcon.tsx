import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { cx } from '@shared/utils/cx';
import { TPlanStatusOption } from '@shared/hooks/usePlanStatusOptions';

type Props = {
  item: TPlanStatusOption;
};

export const PlanGroupProgressIcon = ({ item }: Props) => (
  <IconContainer
    Icon={item.Icon}
    className={cx('bg-white outline outline-1 outline-neutral-300 rounded-xs mr-xxs', {
      'text-neutral-700': item.variant === 'neutral',
      'text-secondary-500': item.variant === 'secondary',
      'text-success-500': item.variant === 'success',
      'text-danger-500': item.variant === 'danger',
    })}
    paddingSize='xxs'
    size='sm'
  />
);
