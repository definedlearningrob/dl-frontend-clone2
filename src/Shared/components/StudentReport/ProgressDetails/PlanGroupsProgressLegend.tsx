import { useTranslation } from 'react-i18next';

import { ReactComponent as CertificateIcon } from '@shared/svg/certificate.svg';
import { PlanStatusLegend } from '@shared/components/PlanStatusLegend/PlanStatusLegend';

export const PlanGroupsProgressLegend = () => {
  const { t } = useTranslation();

  const additionalItems = [
    {
      variant: 'neutral' as const,
      label: t('studentGoalReport.statements'),
      Icon: CertificateIcon,
    },
  ];

  return (
    <div className='bg-neutral-200 p-xs xxxl:p-sm leading-lg rounded-sm mb-base xxxl:mb-md'>
      <PlanStatusLegend additionalItems={additionalItems} />
    </div>
  );
};
