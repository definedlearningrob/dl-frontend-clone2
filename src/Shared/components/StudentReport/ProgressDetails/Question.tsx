import { useTranslation } from 'react-i18next';
import { isEmpty } from 'lodash-es';

import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';
import { ReactComponent as QuestionIcon } from '@shared/svg/question.svg';
import { TPlanGroupStatementQuestion } from '@shared/graphql/user/query/studentPortfolioPlanWithEvaluation';
import { STATEMENT_QUESTION_TYPE } from '@shared/resources/enums';

type Props = {
  question: TPlanGroupStatementQuestion | null;
};

const getAnswer = (question: TPlanGroupStatementQuestion) => {
  const answers = question.answer?.answer || [];

  const selectedOptions = question?.options.filter((option) => answers.includes(option.option));

  return selectedOptions?.map((option) => option.option).join(', ');
};

export const Question = ({ question }: Props) => {
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });
  const { t } = useTranslation();

  if (isEmpty(question)) {
    return null;
  }

  const isTextQuestion =
    question.questionType === STATEMENT_QUESTION_TYPE.LONG_TEXT ||
    question.questionType === STATEMENT_QUESTION_TYPE.SHORT_TEXT;

  const rawAnswer = question?.answer?.answer;

  const answer =
    question && isTextQuestion && !isEmpty(rawAnswer) ? rawAnswer : getAnswer(question);

  return (
    <div>
      <div className='uppercase text-font-primary text-xxs font-bold leading-sm mb-xxs'>
        {question.text}
      </div>
      <div className='mx-sm my-x flex gap-xs items-center'>
        <div>
          {isFullHD ? (
            <div className='h-md flex justify-center items-center rounded-sm text-xs font-medium leading-lg w-[80px] bg-neutral-200 text-neutral-700 group-hover/evidence-item:bg-white'>
              {t('studentGoalReport.answer')}
            </div>
          ) : (
            <IconContainer
              Icon={QuestionIcon}
              className='bg-neutral-200 text-font-secondary rounded-sm'
              paddingSize='xxs'
              size='base'
            />
          )}
        </div>
        <div className='leading-lg text-xxs font-medium'>
          <div className='text-font-primary'>
            {isEmpty(answer) ? (
              <span className='italic text-font-secondary font-medium'>
                {t('studentGoalReport.noAnswer')}
              </span>
            ) : (
              answer
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
