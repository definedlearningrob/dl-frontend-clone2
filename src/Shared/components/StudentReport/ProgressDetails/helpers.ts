import { isEmpty, mapValues } from 'lodash-es';

import { EVALUATION_RESULTS_VALUES } from '@shared/resources/enums';
import { TPlanGroupStatement } from '@shared/graphql/student/query/studentReportProgressByStudent';

export const getGroupProgress = (statements: TPlanGroupStatement[] | undefined = []) => {
  const initialProgress = {
    [EVALUATION_RESULTS_VALUES.COMPLETED]: 0,
    [EVALUATION_RESULTS_VALUES.IN_PROGRESS]: 0,
    [EVALUATION_RESULTS_VALUES.NOT_STARTED]: 0,
    [EVALUATION_RESULTS_VALUES.NOT_MET]: 0,
  };

  if (isEmpty(statements)) {
    return { progress: initialProgress, percentageProgress: initialProgress };
  }

  const progress = statements.reduce((acc, { results }) => {
    if (isEmpty(results)) {
      acc[EVALUATION_RESULTS_VALUES.NOT_STARTED]++;
    } else {
      const latestResult = results[0];
      acc[latestResult.result]++;
    }

    return acc;
  }, initialProgress);

  const percentageProgress = mapValues(progress, (value) => (value / statements.length) * 100);

  return { progress, percentageProgress };
};
