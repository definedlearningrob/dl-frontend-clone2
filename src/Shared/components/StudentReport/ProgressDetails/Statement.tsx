import { useTranslation } from 'react-i18next';
import { isEmpty } from 'lodash-es';

import { Kicker } from '@shared/components/Kicker';
import { StatementStatusBadge } from '@shared/components/StatementStatusBadge';
import { usePlanStatusOptions } from '@shared/hooks/usePlanStatusOptions';
import { EVALUATION_RESULTS_VALUES } from '@shared/resources/enums';
import { formatDateTime } from '@shared/utils/date';
import { Tooltip } from '@shared/components/Tooltip';
import { TPlanGroupStatement } from '@shared/graphql/student/query/studentReportProgressByStudent';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

import { Question } from './Question';
import { EvidenceList } from './EvidenceList';

type Props = {
  statement: TPlanGroupStatement;
};

export const Statement = ({ statement }: Props) => {
  const { t } = useTranslation();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  const { name, isRequired, results, question, evidences } = statement;

  const { getOption } = usePlanStatusOptions();

  const hasResults = !isEmpty(results);

  const latestEvaluationResult = hasResults
    ? results[0]?.result
    : EVALUATION_RESULTS_VALUES.NOT_STARTED;

  const lastUpdated = hasResults ? formatDateTime(results[0].createdAt) : null;

  const statementHasContent = !isEmpty(question) || !isEmpty(evidences);

  return (
    <>
      <div className='bg-neutral-200 rounded-sm xxxl:px-md px-base py-sm flex justify-between'>
        <div>
          {isRequired && (
            <Kicker className='!mb-xxs' size={isFullHD ? 'md' : 'sm'} variant='secondary'>
              {t('sharedCommon.required')}
            </Kicker>
          )}
          <h6 className='font-bold text-xs xxxl:text-sm leading-base mb-xs'>{name}</h6>
          <div className='flex items-center'>
            <div className='text-xs xxxl:text-sm mr-xs font-medium'>
              {t('studentGoalReport.status')}
            </div>
            <StatementStatusBadge
              {...getOption(latestEvaluationResult)!}
              badgeClassName='bg-white'
              size={isFullHD ? 'big' : 'base'}
            />
          </div>
        </div>
        {lastUpdated && (
          <Tooltip
            delayDuration={300}
            message={formatDateTime(lastUpdated, { dateFormat: 'MMMM D, YYYY' })}>
            <div className='text-xxs xxxl:text-xs leading-lg tracking-[0.12px] whitespace-nowrap'>
              {[t('studentGoalReport.lastUpdated'), lastUpdated].join(' ')}
            </div>
          </Tooltip>
        )}
      </div>
      {statementHasContent && (
        <div className='pl-lg xxxl:pl-2lg flex flex-col gap-sm xxxl:gap-base'>
          <Question question={question} />
          <EvidenceList evidence={evidences} />
        </div>
      )}
    </>
  );
};
