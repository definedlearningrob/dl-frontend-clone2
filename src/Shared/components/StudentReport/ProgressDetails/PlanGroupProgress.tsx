import { ReactComponent as CertificateIcon } from '@shared/svg/certificate.svg';
import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { cx } from '@shared/utils/cx';
import { usePlanStatusOptions } from '@shared/hooks/usePlanStatusOptions';
import { getGroupProgress } from '@shared/components/StudentReport/ProgressDetails/helpers';
import { PlanGroupProgressIcon } from '@shared/components/StudentReport/ProgressDetails/PlanGroupProgressIcon';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';

type Props = {
  className?: string;
  progressData: ReturnType<typeof getGroupProgress>;
  isLoading: boolean;
};

const commonMarkerClassName = cx(
  'flex items-center justify-center',
  'relative after:absolute pe-[20px] xxxl:pe-[28px] after:bg-neutral-300',
  'after:h-xxs after:w-xxs after:rounded-full after:right-xs xxxl:after:right-x',
  'last-of-type:after:hidden last-of-type:pe-0'
);

export const PlanGroupProgress = ({ progressData, isLoading }: Props) => {
  const { options } = usePlanStatusOptions();

  const { progress, percentageProgress } = progressData;

  if (isLoading) {
    return (
      <div className='flex'>
        {options.map((item, index) => (
          <div key={index} className={commonMarkerClassName}>
            <PlanGroupProgressIcon item={item} />
            <div className='w-lg'>
              <SkeletonRectangle size='full-width' />
            </div>
          </div>
        ))}
      </div>
    );
  }

  return (
    <div className='flex'>
      {options.map((item, index) => {
        const value = progress[item.status];
        const percentageValue = Math.round(percentageProgress[item.status]);

        return (
          <div key={index} className={commonMarkerClassName}>
            <PlanGroupProgressIcon item={item} />
            <div className='font-medium leading-lg mr-xxxs text-xxs xxxl:text-xs'>
              {`${percentageValue}`}%
            </div>
            <div className='text-xxs xxxl:text-xs leading-sm flex items-center text-font-secondary'>
              (
              <IconContainer
                Icon={CertificateIcon}
                className='mr-xxxs'
                paddingSize='none'
                size='x'
              />
              {value})
            </div>
          </div>
        );
      })}
    </div>
  );
};
