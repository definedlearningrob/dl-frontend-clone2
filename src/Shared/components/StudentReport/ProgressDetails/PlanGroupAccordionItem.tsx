import { AnimatePresence, motion } from 'framer-motion';
import { useTranslation } from 'react-i18next';
import * as Accordion from '@radix-ui/react-accordion';
import { useMemo } from 'react';
import { sortBy } from 'lodash-es';

import { PieChart } from '@shared/components/PieChart';
import { cx } from '@shared/utils/cx';
import { ReactComponent as ChevronUpIcon } from '@shared/svg/chevron_up.svg';
import { ReactComponent as ChevronDownIcon } from '@shared/svg/chevron_down.svg';
import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { PlanGroupProgress } from '@shared/components/StudentReport/ProgressDetails/PlanGroupProgress';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';
import { InjectedContent } from '@shared/components/IncjectedContent/InjectedContent';
import { getGroupProgress } from '@shared/components/StudentReport/ProgressDetails/helpers';
import { EVALUATION_RESULTS_VALUES } from '@shared/resources/enums';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';
import { TPlanGroup } from '@shared/graphql/student/query/studentReportProgressByStudent';
import { IconButton } from '@shared/components/IconButton/IconButton';

import { Statement } from './Statement';

type Props = {
  isExpanded: boolean;
  group?: TPlanGroup;
  isLoading: boolean;
};

export const PlanGroupAccordionItem = ({ isExpanded, group, isLoading }: Props) => {
  const { t } = useTranslation();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });

  const { displayName, name, statements, description } = group || {};

  const sortedStatements = useMemo(() => sortBy(statements, 'name'), [statements]);

  const chartWrapperClassName = cx('transition-all ease-out p-xxs shrink-0', {
    'w-[132px] h-[132px] xxxl:w-[172px] xxxl:h-[172px] rotate-[360deg] duration-[500ms]':
      isExpanded,
    'w-[56px] h-[56px] rotate-[0deg] duration-[300ms]': !isExpanded,
  });

  const planGroupClassName = cx(
    'relative rounded-sm border border-neutral-300 p-sm xxxl:p-base',
    'focus-visible:!border-secondary-500 focus-visible:border',
    {
      'group hover:bg-neutral-200': !isExpanded && !isLoading,
      'border-primary-500 mb-sm xxxl:mb-base': isExpanded,
    }
  );

  const chevronClassName = cx(
    '!bg-white transition-colors',
    '!rounded-sm !border-white focus-visible:!border-primary-500',
    {
      'rotate-180': isExpanded,
    }
  );

  const progressData = useMemo(() => getGroupProgress(statements), [statements]);

  const chartData = [
    {
      name: EVALUATION_RESULTS_VALUES.NOT_STARTED,
      value: progressData.progress.NOT_STARTED,
      colorClassName: 'fill-neutral-600',
    },
    {
      name: EVALUATION_RESULTS_VALUES.IN_PROGRESS,
      value: progressData.progress.IN_PROGRESS,
      colorClassName: 'fill-secondary-500',
    },
    {
      name: EVALUATION_RESULTS_VALUES.COMPLETED,
      value: progressData.progress.COMPLETED,
      colorClassName: 'fill-success-500',
    },
    {
      name: EVALUATION_RESULTS_VALUES.NOT_MET,
      value: progressData.progress.NOT_MET,
      colorClassName: 'fill-danger-500',
    },
  ];

  return (
    <div className={planGroupClassName}>
      <Accordion.Header asChild={true}>
        <Accordion.Trigger asChild={true}>
          <button
            className={cx(
              'text-start w-full flex gap-base xxxl:gap-md focus-visible:outline-offset-4',
              {
                'mb-base xxxl:mb-md': isExpanded,
                'cursor-pointer': !isLoading,
              }
            )}>
            <div className={chartWrapperClassName}>
              {isLoading && <SkeletonRectangle height='full-height' size='full-width' />}
              {!isLoading && <PieChart data={chartData} isResponsive={true} />}
            </div>
            <div className='w-full focus-visible:!outline-secondary-500'>
              <div
                className={cx('flex gap-base transition-[margin]', {
                  'mb-sm xxxl:mb-base': isExpanded,
                })}>
                <div className='mr-auto'>
                  <h6 className='font-bold leading-base mb-xs max-w-[640px] xxxl:max-w-[840px]'>
                    {isLoading ? (
                      <div className='w-[120px]'>
                        <SkeletonRectangle size='full-width' />
                      </div>
                    ) : (
                      displayName || name
                    )}
                  </h6>
                  <div>
                    <PlanGroupProgress isLoading={isLoading} progressData={progressData} />
                  </div>
                </div>
                <div className='flex items-center gap-sm xxxl:gap-base'>
                  <div className='flex justify-center items-center font-medium text-xxs xxxl:text-xs'>
                    <div className='me-xxs xxxl:me-xs'>
                      {t('studentGoalReport.statementsCount')}
                    </div>
                    {isLoading ? (
                      <div className='w-base h-base'>
                        <SkeletonRectangle height='full-height' size='full-width' />
                      </div>
                    ) : (
                      <div className='flex justify-center items-center bg-neutral-200 group-hover:bg-white rounded-full min-w-[24px] min-h-[24px] xxxl:min-w-[32px] xxxl:min-h-[32px] px-xs py-xxs'>
                        {statements?.length || 0}
                      </div>
                    )}
                  </div>
                  <IconContainer
                    Icon={ChevronDownIcon}
                    className={chevronClassName}
                    paddingSize={isFullHD ? 'xs' : 'xxs'}
                    size='base'
                  />
                </div>
              </div>
              <AnimatePresence>
                {description && isExpanded && (
                  <motion.div
                    key='content'
                    animate='open'
                    className='flex flex-col gap-xs xxxl:gap-sm max-w-[680px] xxxl:max-w-[840px]'
                    exit='collapsed'
                    initial='collapsed'
                    transition={{
                      duration: 0.3,
                      opacity: { duration: 0.1, delay: 0.1 },
                    }}
                    variants={{
                      open: {
                        height: 'auto',
                        opacity: 1,
                      },
                      collapsed: { height: 0, opacity: 0 },
                    }}>
                    <div className='font-bold text-xxxs xxxl:text-xxs uppercase'>
                      {t('studentGoalReport.description')}
                    </div>
                    <div className='text-font-secondary leading-lg text-xxs xxxl:text-xs'>
                      <InjectedContent content={description} />
                    </div>
                  </motion.div>
                )}
              </AnimatePresence>
            </div>
          </button>
        </Accordion.Trigger>
      </Accordion.Header>
      <Accordion.Content>
        <AnimatePresence>
          {statements && isExpanded && (
            <motion.div
              key='content'
              animate='open'
              className='flex flex-col gap-base xxxl:gap-md'
              exit='collapsed'
              initial='collapsed'
              transition={{
                duration: 0.3,
                opacity: { duration: 0.1, delay: 0.1 },
              }}
              variants={{
                open: {
                  height: 'auto',
                  opacity: 1,
                },
                collapsed: { height: 0, opacity: 0 },
              }}>
              {sortedStatements.map((statement, index) => (
                <Statement key={`${index}-${statement.name}`} statement={statement} />
              ))}
            </motion.div>
          )}
        </AnimatePresence>
      </Accordion.Content>
      {isExpanded && (
        <Accordion.Trigger asChild={true}>
          <IconButton
            Icon={ChevronUpIcon}
            circle={true}
            className='absolute bottom-0 left-1/2 translate-y-1/2 -translate-x-1/2 bg-white'
            size={isFullHD ? 'lg' : 'md'}
            variant='primary-outlined'
          />
        </Accordion.Trigger>
      )}
    </div>
  );
};
