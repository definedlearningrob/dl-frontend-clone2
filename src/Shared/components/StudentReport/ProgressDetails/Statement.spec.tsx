import { screen } from '@testing-library/react';

import { Statement } from '@shared/components/StudentReport/ProgressDetails/Statement';
import {
  EVALUATION_RESULTS_VALUES,
  EVIDENCE_CONTEXT_KIND,
  EVIDENCE_KIND,
  SERVICE_NAME,
  STATEMENT_QUESTION_TYPE,
} from '@shared/resources/enums';
import { renderWithI18N } from '@shared/utils/renderWithI18N';
import { TPlanGroupStatement } from '@shared/graphql/student/query/studentReportProgressByStudent';

describe('Statement component', () => {
  it('should render required statement correctly', () => {
    const statement = {
      name: 'Test statement',
      isRequired: true,
      results: [],
      question: null,
      evidences: [],
      step: 1,
      id: '3',
    };

    const { container } = renderWithI18N(<Statement statement={statement} />, 'DC');
    expect(screen.getByText('Required')).toBeInTheDocument();
    expect(screen.getByText('Test statement')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render statement with results correctly', () => {
    const statement = {
      name: 'Test statement',
      isRequired: false,
      results: [
        { result: EVALUATION_RESULTS_VALUES.COMPLETED, createdAt: '2021-09-01' },
        { result: EVALUATION_RESULTS_VALUES.NOT_STARTED, createdAt: '2021-09-01' },
      ],
      question: null,
      evidences: [],
      step: 1,
      id: '1',
    };

    const { container } = renderWithI18N(<Statement statement={statement} />, 'DC');
    expect(screen.getByText('Completed')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render statement with question correctly', () => {
    const statement = {
      id: '4',
      step: 6,
      name: 'Test statement',
      isRequired: false,
      results: [],
      question: {
        id: '213',
        text: 'Test question',
        questionType: STATEMENT_QUESTION_TYPE.SHORT_TEXT,
        answer: { answer: ['Sample answer'], id: '1' },
        options: [],
      },
      evidences: [],
    };

    const { container } = renderWithI18N(<Statement statement={statement} />, 'DC');
    expect(screen.getByText('Test question')).toBeInTheDocument();
    expect(screen.getByText('Sample answer')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render statement with evidences correctly', () => {
    const statement: TPlanGroupStatement = {
      id: '4',
      step: 6,
      name: 'Test statement',
      isRequired: false,
      results: [],
      question: null,
      evidences: [
        {
          label: 'Schedule 1',
          itemId: '1',
          type: EVIDENCE_KIND.ASSIGNMENT_SUBMISSION,
          contextType: EVIDENCE_CONTEXT_KIND.COURSE,
          service: SERVICE_NAME.CAREERS,
          updatedAt: '2023-12-12T11:05:56Z' as unknown as Date,
          isTeamSubmission: false,
          id: '191',
          rubricScores: null,
        },
        {
          label: 'Software Bug Feedback Form Assignment',
          type: EVIDENCE_KIND.ASSIGNMENT_SUBMISSION,
          contextType: EVIDENCE_CONTEXT_KIND.VIRTUAL_INTERNSHIP,
          service: SERVICE_NAME.CAREERS,
          itemId: '2',
          updatedAt: '2023-08-31T06:52:39Z' as unknown as Date,
          isTeamSubmission: false,
          id: '192',
          rubricScores: null,
        },
        {
          label: 'Schedule 2',
          type: EVIDENCE_KIND.RUBRIC_GRADE,
          contextType: null,
          service: SERVICE_NAME.CAREERS,
          updatedAt: '2023-12-12T11:05:56Z' as unknown as Date,
          itemId: '3',
          isTeamSubmission: false,
          id: null,
          rubricScores: [
            {
              label: 'Sample rubric',
              maxScore: 25,
              currentScore: 12,
            },
          ],
        },
      ],
    };

    const { container } = renderWithI18N(<Statement statement={statement} />, 'DC');
    expect(screen.getByText('Sample rubric')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
