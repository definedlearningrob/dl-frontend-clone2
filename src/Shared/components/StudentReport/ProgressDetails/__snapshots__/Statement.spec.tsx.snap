// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statement component should render required statement correctly 1`] = `
<div>
  <div
    class="bg-neutral-200 rounded-sm xxxl:px-md px-base py-sm flex justify-between"
  >
    <div>
      <div
        class="kicker sm secondary !mb-xxs"
      >
        Required
      </div>
      <h6
        class="font-bold text-xs xxxl:text-sm leading-base mb-xs"
      >
        Test statement
      </h6>
      <div
        class="flex items-center"
      >
        <div
          class="text-xs xxxl:text-sm mr-xs font-medium"
        >
          Status:
        </div>
        <div
          class="font-medium leading-lg flex items-center inline-flex items-center gap-xxs bg-white bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          Not started
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Statement component should render statement with evidences correctly 1`] = `
<div>
  <div
    class="bg-neutral-200 rounded-sm xxxl:px-md px-base py-sm flex justify-between"
  >
    <div>
      <h6
        class="font-bold text-xs xxxl:text-sm leading-base mb-xs"
      >
        Test statement
      </h6>
      <div
        class="flex items-center"
      >
        <div
          class="text-xs xxxl:text-sm mr-xs font-medium"
        >
          Status:
        </div>
        <div
          class="font-medium leading-lg flex items-center inline-flex items-center gap-xxs bg-white bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          Not started
        </div>
      </div>
    </div>
  </div>
  <div
    class="pl-lg xxxl:pl-2lg flex flex-col gap-sm xxxl:gap-base"
  >
    <div
      class="flex flex-col gap-xxs xxxl:gap-xs"
    >
      <h5
        class="mb-0 text-xxs leading-base uppercase"
      >
        Evidence 
        <span
          class="text-neutral-600"
        >
          (3)
        </span>
        :
      </h5>
      <ul>
        <li
          class="mx-sm my-x"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="flex gap-xs items-start"
            >
              <div
                class="h-md flex justify-center items-center rounded-sm text-xs font-medium leading-lg group-hover/evidence-item:bg-white w-md bg-danger-100 text-danger-500"
              >
                DC
              </div>
              <div
                class="leading-lg text-xxs font-medium"
              >
                <div
                  class="text-primary-500 flex gap-xxs items-center"
                >
                  Schedule 1
                </div>
                <div
                  class="text-neutral-800 flex relative"
                >
                  <div
                    class="relative after:absolute pe-x xxxl:pe-[20px] after:bg-neutral-300 after:h-xxs after:w-xxs after:rounded-full after:right-xxs xxxl:after:right-xs after:top-xs"
                  >
                    Product
                  </div>
                  <span
                    data-state="closed"
                  >
                    <span
                      class="text-neutral-600"
                    >
                      Dec 12, 2023
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="flex gap-xs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-sm group-hover/evidence-item:bg-white bg-success-100 text-success-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
        </li>
        <li
          class="mx-sm my-x"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="flex gap-xs items-start"
            >
              <div
                class="h-md flex justify-center items-center rounded-sm text-xs font-medium leading-lg group-hover/evidence-item:bg-white w-md bg-danger-100 text-danger-500"
              >
                DC
              </div>
              <div
                class="leading-lg text-xxs font-medium"
              >
                <div
                  class="text-primary-500 flex gap-xxs items-center"
                >
                  Software Bug Feedback Form Assignment
                </div>
                <div
                  class="text-neutral-800 flex relative"
                >
                  <div
                    class="relative after:absolute pe-x xxxl:pe-[20px] after:bg-neutral-300 after:h-xxs after:w-xxs after:rounded-full after:right-xxs xxxl:after:right-xs after:top-xs"
                  >
                    Virtual Internship
                  </div>
                  <span
                    data-state="closed"
                  >
                    <span
                      class="text-neutral-600"
                    >
                      Aug 31, 2023
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="flex gap-xs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-sm group-hover/evidence-item:bg-white bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
        </li>
        <li
          class="mx-sm my-x"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="flex gap-xs items-start"
            >
              <div
                class="h-md flex justify-center items-center rounded-sm text-xs font-medium leading-lg group-hover/evidence-item:bg-white w-md bg-danger-100 text-danger-500"
              >
                DC
              </div>
              <div
                class="leading-lg text-xxs font-medium"
              >
                <div
                  class="text-primary-500 flex gap-xxs items-center"
                >
                  Schedule 2
                </div>
                <div
                  class="text-neutral-800 flex relative"
                >
                  <div
                    class="relative after:absolute pe-x xxxl:pe-[20px] after:bg-neutral-300 after:h-xxs after:w-xxs after:rounded-full after:right-xxs xxxl:after:right-xs after:top-xs"
                  >
                    Rubric row
                  </div>
                  <span
                    data-state="closed"
                  >
                    <span
                      class="text-neutral-600"
                    >
                      Dec 12, 2023
                    </span>
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <div
                    class="relative after:absolute pe-x xxxl:pe-[20px] after:bg-neutral-300 after:h-xxs after:w-xxs after:rounded-full after:right-xxs xxxl:after:right-xs flex items-center after:top-[11px] last-of-type:after:invisible last-of-type:pe-0"
                  >
                    <div
                      class="bg-success-100 !text-success-500 py-xxs px-xs rounded-full me-xxs group-hover/evidence-item:bg-white transition-colors text-xxs"
                    >
                      12/25
                    </div>
                    Sample rubric
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex gap-xs"
            >
              <span
                data-state="closed"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 rounded-sm group-hover/evidence-item:bg-white bg-primary-200 text-primary-500 p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Statement component should render statement with question correctly 1`] = `
<div>
  <div
    class="bg-neutral-200 rounded-sm xxxl:px-md px-base py-sm flex justify-between"
  >
    <div>
      <h6
        class="font-bold text-xs xxxl:text-sm leading-base mb-xs"
      >
        Test statement
      </h6>
      <div
        class="flex items-center"
      >
        <div
          class="text-xs xxxl:text-sm mr-xs font-medium"
        >
          Status:
        </div>
        <div
          class="font-medium leading-lg flex items-center inline-flex items-center gap-xxs bg-white bg-neutral-200 text-font-secondary rounded-sm p-xxs text-xs gap-xxs"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          Not started
        </div>
      </div>
    </div>
  </div>
  <div
    class="pl-lg xxxl:pl-2lg flex flex-col gap-sm xxxl:gap-base"
  >
    <div>
      <div
        class="uppercase text-font-primary text-xxs font-bold leading-sm mb-xxs"
      >
        Test question
      </div>
      <div
        class="mx-sm my-x flex gap-xs items-center"
      >
        <div>
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0 bg-neutral-200 text-font-secondary rounded-sm p-xxs"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
        </div>
        <div
          class="leading-lg text-xxs font-medium"
        >
          <div
            class="text-font-primary"
          >
            Sample answer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Statement component should render statement with results correctly 1`] = `
<div>
  <div
    class="bg-neutral-200 rounded-sm xxxl:px-md px-base py-sm flex justify-between"
  >
    <div>
      <h6
        class="font-bold text-xs xxxl:text-sm leading-base mb-xs"
      >
        Test statement
      </h6>
      <div
        class="flex items-center"
      >
        <div
          class="text-xs xxxl:text-sm mr-xs font-medium"
        >
          Status:
        </div>
        <div
          class="font-medium leading-lg flex items-center inline-flex items-center gap-xxs bg-white bg-success-100 text-success-500 rounded-sm p-xxs text-xs gap-xxs"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          Completed
        </div>
      </div>
    </div>
    <span
      data-state="closed"
    >
      <div
        class="text-xxs xxxl:text-xs leading-lg tracking-[0.12px] whitespace-nowrap"
      >
        Last updated: Sep 1, 2021
      </div>
    </span>
  </div>
</div>
`;
